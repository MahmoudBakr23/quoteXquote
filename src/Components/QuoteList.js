/* eslint-disable react/no-array-index-key */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { saveQuote } from '../Actions/saveQuote';
import { changeChar } from '../Actions/search';
import Search from './Search';

const QuoteList = () => {
  const params = useParams();
  const dispatch = useDispatch();

  const getData = async () => {
    const response = await fetch(`https://animechan.vercel.app/api/quotes/anime?title=${params.anime}`);
    const myData = await response.json();

    dispatch(saveQuote(myData));
  };

  const animes = useSelector((state) => state);

  useEffect(() => {
    getData();
    animes.searchReducer.name = '';
  }, []);

  const handleSearch = (e) => {
    dispatch(changeChar(e.target.value));
  };

  const quotesDisplay = () => {
    if (animes.searchReducer.name === '') {
      return animes.quoteReducer;
    }
    return animes.quoteReducer.filter((quote) => quote.character === animes.searchReducer.name);
  };

  return (
    <div className="anime-list">
      <Search handleSearch={handleSearch} />
      <h3>{params.anime}</h3>
      {quotesDisplay().map((quote, index) => (
        <div key={index}>
          <h5>{quote.character}</h5>
          <h4>{quote.quote}</h4>
        </div>
      ))}
    </div>
  );
};

export default QuoteList;
