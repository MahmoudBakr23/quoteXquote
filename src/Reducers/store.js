import { createStore } from 'redux';
import animeReducer from './anime';

const store = createStore(animeReducer);

export default store;
