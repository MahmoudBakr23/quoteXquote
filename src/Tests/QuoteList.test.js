import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import QuoteList from '../Components/QuoteList';
import store from '../Reducers/store';

it('Expect to render Quote List component', () => {
  const result = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <QuoteList />
      </BrowserRouter>
    </Provider>,
  ).toJSON;
  expect(result).toMatchSnapshot();
});
