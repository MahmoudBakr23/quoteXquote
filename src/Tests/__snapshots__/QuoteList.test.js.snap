// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Quote List component 1`] = `[Function]`;
